v2/pkg/runner/options.go

```diff
@@ -203,27 +193,28 @@ func ParseOptions() *Options {
 	return options
 }
 
-func migrateToProviderConfig(source, dest string) error {
-	fileSource, err := os.Open(source)
-	if err != nil {
-		return err
+// loadProvidersFrom runs the app with source config
+func (options *Options) loadProvidersFrom(location string) {
+	// todo: move elsewhere
+	if len(options.Resolvers) == 0 {
+		options.Resolvers = resolve.DefaultResolvers
 	}
-	defer fileSource.Close()
-
-	providers := &Providers{}
 
-	// create empty template at destination, so in case of failure, the file is generated
-	if err := providers.MarshalTo(dest); err != nil {
-		return err
+	// We skip bailing out if file doesn't exist because we'll create it
+	// at the end of options parsing from default via goflags.
+	if err := UnmarshalFrom(location); isFatalErr(err) && !errors.Is(err, os.ErrNotExist) {
+		gologger.Fatal().Msgf("Could not read providers from '%s': %s\n", location, err)
 	}
+}
 
-	// unmarshal fields to migrate into temporary struct
-	if err := yaml.NewDecoder(fileSource).Decode(providers); isFatalErr(err) {
+func migrateToProviderConfig(source, dest string) error {
+	fileSource, err := os.Open(source)
```

Comment:
Previously `source` was unmarshaled into a temporary providers struct, then marshaled to the destination. Here it seems it's not used anymore?

Reply:
![image](https://user-images.githubusercontent.com/13679401/185992814-d0c21531-ee39-4a95-b9f7-88754c88c96c.png)